File: app/src/main/java/vedder/xander/brewtracker/model/Recipe.java
package vedder.xander.brewtracker.model;

import java.lang.reflect.Field;
import java.time.LocalDate;

public class Recipe extends AbstractDataItem {

    private LocalDate createdAt;
    private String name;
    private String type;

    public Recipe(LocalDate createdAt, String name, String type) {
        this.createdAt = createdAt;
        this.name = name;
        this.type = type;
    }

    public LocalDate getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDate createdAt) {
        this.createdAt = createdAt;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String get(String parameterName) {
        Field[] fields = this.getClass().getDeclaredFields();
        for (Field field : fields) {
            if (field.getName().equals(parameterName)) {
                try {
                    return field.get(this).toString();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
        return null;
    }
}

File: app/src/main/java/vedder/xander/brewtracker/recipe/model/Recipe.kt
package vedder.xander.brewtracker.recipe.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import vedder.xander.brewtracker.brew.model.BrewType
import vedder.xander.brewtracker.ingredient.model.Ingredient
import java.time.LocalDate

@Parcelize
data class Recipe(
        var createdAt: LocalDate,
        var name: String,
        var type: BrewType,
        var ingredients: MutableList<Ingredient>
) : Parcelable
